app/models/druid_version_zip.rb

```diff
@@ -26,10 +27,63 @@ def file_path
     File.join(Settings.zip_storage, s3_key)
   end
 
-  # Currently computed live, soon will fetch from checksum storage
-  # @return [String] md5 base64-encoded checksum
-  # @todo Fetch (and cache) md5 from checksum storage
+  # WITHOUT (re)digesting the file, convert a hexdigest MD5 value to base64-endcoded equivalent.
+  # Motivation: we store hexdigest and S3 requires base64 in headers.
+  # @param [String] hex
+  # @return [String] base64 equivalent
+  def hex_to_base64(hex)
+    [[hex].pack("H*")].pack("m0")
+  end
+
+  # @return [Digest::MD5] cached md5 object
   def md5
-    @md5 ||= Digest::MD5.file(file_path).base64digest
+    @md5 ||= Digest::MD5.file(file_path)
+  end
+
+  # @return [Hash<Symbol => [String, Integer]>] metadata to accompany Zip file to an endpoint
+  def metadata
+    { checksum_md5: hexdigest, size: size, zip_cmd: zip_command, zip_version: zip_version }
+  end
+
+  def moab_version_path
+    @moab_version_path ||= Moab::StorageServices.object_version_path(druid.id, version)
+  end
+
+  # @return [Integer] Zip file size
+  def size
+    @size ||= FileTest.size(file_path)
+  end
+
+  # @return [String] shell command to unzip
+  # def unzip_command
+  #   "unzip #{file_path} -d #{place_to_unzip}"
+  # end
+
+  # @return [String] shell command to create this zip
+  def zip_command
+    "zip -vr0X -s 10g #{file_path} #{moab_version_path}"
+  end
+
+  # We presume the system guts do not change underneath a given class instance.
+  # We want to avoid shelling out (forking) unnecessarily, just for the version.
+  def zip_version
```

Comment:
The moab version in this model is `version` (as in the initialization).

The value from this method is extracted from `zip -v`, hence `zip_version`.  I get that we are talking about different zips (files vs. the util) and different versions (moab int vs. util semver/date), but that problem is sorta inherent.  `zip_command_version` is more confusing to me, because it seems like it would be if we change the structure of `zip_command` and wanted to track versions of that.  I'm open to some other better name, I just didn't find any I preferred.

Reply:
perhaps @jmartin-sul (our naming expert, per Sandi Metz) has an idea.  🤣

Reply:
I would also prefer "operating system" in comments  - dunno why that tripped me up, but it did.

Reply:
Maybe `zip_util_version`?